src/engine/SCons/Spirent/__init__.py

```diff
@@ -148,7 +149,10 @@ def _walk_tree_write_nodes(self, f, parent_id, node, visited, written):
                 target = [node]
                 source = node.sources
                 env = node.env or node.builder.env
-                builder_id = abs(hash(str(act.get_presig(target, source, env))))
+                # Get the contents of the action command line and hash it
+                from SCons.Subst import SUBST_CMD
```

Comment:
Since you're importing `SCons.Subst` at the top of the file you should be able to use `SCons.Subst.SUBST_CMD` when needed. Or change the import at the top to `from SCons.Subst import SUBST_CMD`. You shouldn't need both.